OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -34,7 +34,7 @@ public Attack(Actor self, Target target, bool allowMovement, bool forceAttack)
 
 			this.forceAttack = forceAttack;
 
-			attack = self.Trait<AttackBase>();
+			attack = self.TraitsImplementing<AttackBase>().First(x => !x.IsTraitDisabled);
```

Comment:
Good point.
Perhaps `attack` should be initialized (`attack = self.TraitsImplementing<AttackBase>().FirstOrDefault(x => !x.IsTraitDisabled)`) on first `Tick` and the activity canceled if `attack == null`.